js/src/main/kotlin/org/rust/js/RsWasmBindgenLineMarkerProvider.kt

```diff
@@ -0,0 +1,91 @@
+import com.intellij.psi.util.nextLeaf
+import com.intellij.util.ConstantFunction
+import icons.JavaScriptPsiIcons
+import org.rust.lang.core.crate.crateGraph
+import org.rust.lang.core.psi.RsEnumItem
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.lang.core.psi.ext.RsOuterAttributeOwner
+import org.rust.lang.core.psi.ext.RsVisibilityOwner
+import org.rust.lang.core.psi.ext.containingCargoPackage
+import org.rust.lang.core.psi.ext.findOuterAttr
+import org.rust.lang.core.psi.impl.RsBaseTypeImpl
+import java.nio.file.Paths
+
+class RsWasmBindgenLineMarkerProvider : RelatedItemLineMarkerProvider() {
+    override fun getLineMarkerInfo(element: PsiElement): RelatedItemLineMarkerInfo<*>? {
+        // Line marker will be available only if there is wasm-bindgen dependency in the project,
+        // current element is public and can have outer attribute
+        if (element.project.crateGraph.topSortedCrates.none { it.normName == "wasm_bindgen" }) return null
+        if (element !is RsOuterAttributeOwner) return null
+        if (element is RsVisibilityOwner && element !is RsImplItem && !element.isPublic) return null
+
+        // Check if attribute is basic #[wasm_bindgen] attribute
+        val attr = element.findOuterAttr("wasm_bindgen")
+            ?.nextLeaf() ?: return null
+
+        val cargoPackage = element.containingCargoPackage ?: return null
+        val expectedName = cargoPackage.normName
+        val basePath = cargoPackage.contentRoot?.path ?: return null
+
+        // Expecting to find related generated typescript declarations in pkg/ folder of current package
+        val expectedPath = Paths.get(basePath, "pkg/$expectedName.d.ts").toString()
+        val tsFile = LocalFileSystem.getInstance().findFileByPath(expectedPath) ?: return null
+        val tsPsiFile = PsiManager.getInstance(element.project).findFile(tsFile) ?: return null
+
+        // Trying to find the correlating element with the same name
+        val destination = when (element) {
+            // If it is impl, then resolving to the related struct and using its name
+            is RsImplItem -> {
+                val reference = (element.typeReference as? RsBaseTypeImpl) ?: return null
+                val struct = (reference.path?.reference?.resolve() as? RsStructItem) ?: return null
+                findRelatedTsElement(struct, tsPsiFile)
+            }
+            else -> findRelatedTsElement(element, tsPsiFile)
+        } ?: return null
+
+        // BACKCOMPAT: 2020.1
+        @Suppress("DEPRECATION")
+        return RelatedItemLineMarkerInfo(
+            attr,
+            attr.textRange,
+            JavaScriptPsiIcons.FileTypes.TypeScriptFile,
+            Pass.LINE_MARKERS,
+            ConstantFunction<PsiElement, String>("Related generated declaration"),
+            DefaultGutterIconNavigationHandler(listOf(destination), "Related generated declaration"),
```

Comment:
```suggestion
            DefaultGutterIconNavigationHandler(listOf(destination), "Generated declarations"),
```
AFAIU this title will be used in case of multiple targets (declarations). Currently, there is always only one target due to `findDescendantOfType`, but maybe there will be several ones later